heed/src/env.rs

```diff
@@ -552,7 +599,12 @@ impl Env {
 
         // safety: The name cstring is cloned by LMDB, we can drop it after.
         //         If a read-only is used with the MDB_CREATE flag, LMDB will throw an error.
-        unsafe { mdb_result(ffi::mdb_dbi_open(raw_txn, name_ptr, flags, &mut dbi))? };
+        unsafe {
+            mdb_result(ffi::mdb_dbi_open(raw_txn, name_ptr, flags, &mut dbi))?;
+            if TypeId::of::<C>() != TypeId::of::<NoopComparator>() {
```

Comment:
Ok, that's smart! I was thinking about the way you were ignoring the `NoopComparator`.

Reply:
Thanks!